Source/MQTTnet/IPayloadSegmentable.cs

```diff
@@ -0,0 +1,21 @@
+namespace MQTTnet
+{
+    public interface IPayloadSegmentable
+    {
+        /// <summary>
+        /// Get or set Payload array
+        /// </summary>
+        byte[] Payload { get; set; }
+
+        /// <summary>
+        /// Get or set the offset of Payload
+        /// </summary>
+        int PayloadOffset { get; set; }
+
+        /// <summary>
+        /// Get or set the effective number of bytes of Payload
+        /// Leaving null means equal to the total length of Payload minus PayloadOffset
+        /// </summary>
+        int? PayloadCount { get; set; }
```

Comment:
```suggestion
        int? PayloadLength { get; set; }
```
is more .NET-like. I associated _Count_ with enumerating, so "how many payloads are there?" not as "how **long** is the payload".

Reply:
Yes, I agree with you